tests/test_topology.py

```diff
@@ -106,6 +106,48 @@ def assert_variants_equal(self, ts1, ts2):
         v2 = list(ts2.variants(as_bytes=True))
         self.assertEqual(v1, v2)
 
+    def check_num_leaves(self, ts, x):
+        """
+        Compare against x, a list of tuples of the form
+            (tree number, parent, number of leaves)
```

Comment:
Minor nitpick here, but I don't think multiline docstrings should be indented like this. It may even break downstream processing tools (like sphinx) since this indentation is significant. See https://www.python.org/dev/peps/pep-0257/#handling-docstring-indentation

Also (less importantly), I think pep8 says that multiline block comments shouldn't be indented either:

> Comments generally apply to some (or all) code that follows them, and are indented to the same level as that code. Each line of a block comment starts with a # and a single space (unless it is indented text inside the comment).

< /nitpick >